dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/NiNumberPage.cshtml.cs

```diff
@@ -0,0 +1,54 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using TeacherIdentity.AuthServer.Pages.SignIn.Trn;
+
+namespace TeacherIdentity.AuthServer.Pages.SignIn.Register;
+
+[BindProperties]
+[RequiresTrnLookup]
+public class NiNumberPage : TrnLookupPageModel
+{
+    public NiNumberPage(IdentityLinkGenerator linkGenerator, TrnLookupHelper trnLookupHelper)
+        : base(linkGenerator, trnLookupHelper)
+    {
+    }
+
+    [Display(Name = "What is your National Insurance number?", Description = "It’s on your National Insurance card, benefit letter, payslip or P60. For example, ‘QQ 12 34 56 C’.")]
+    [Required(ErrorMessage = "Enter a National Insurance number")]
+    [RegularExpression(@"(?i)\A[a-z]{2}(?: [0-9]{2}){3} [a-d]{1}|[a-z]{2}[0-9]{6}[a-d]{1}\Z", ErrorMessage = "Enter a National Insurance number in the correct format")]
```

Comment:
I think this might be case sensitively matching, and requires spaces? Can we normalise for casing and spaces?

We should _probably_ store without spaces - it think they're in the example for readability only?

Reply:
@edwardhorsford  We don't store this beyond the session and when we match with DQT it's normalised to remove spaces.